src/main/java/hudson/plugins/im/GroupChatIMMessageTarget.java

```diff
@@ -57,15 +65,24 @@ public String getName() {
 
     @Deprecated
     public String getPassword() {
-        return password;
+        return Secret.toString(password);
     }
 
     public Secret getSecretPassword() {
-        return secretPassword;
+        return password;
+    }
+
+    @DataBoundSetter
+    public void setSecretPassword(Secret secretPassword) {
+        if (Secret.toString(secretPassword).equals("")) {
```

Comment:
stapler submits empty string which makes it look like this has a password, meaning it shows up in jcasc and in the configure ui as with a password.

with this fix that doesn't happen

```
  ircPublisher:
    charset: "UTF-8"
    commandPrefix: "!jenkins"
    defaultTargets:
    - groupChat:
        name: "#hello"
        notificationOnly: false
    - groupChat:
        name: "#private"
        notificationOnly: false
        secretPassword: "{AQAAABAAAAAQXRHSKt9oLho8NeWn1Yt1qhlsYvvrZ4b58vJ3W4l69tk=}"
```